src/robot/libraries/BuiltIn.py

```diff
@@ -745,87 +761,154 @@ def should_be_equal_as_numbers(self, first, second, msg=None, values=True,
         """Fails if objects are equal after converting them to strings.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``first`` and ``second`` should be compared case-insensitively,
+        provided that ``first`` and ``second`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
         """
         self._log_types_at_info_if_different(first, second)
         first, second = [self._convert_to_string(i) for i in (first, second)]
+        if is_truthy(ignore_case) and is_string(first) and is_string(second):
+            first = first.lower()
+            second = second.lower()
         self._should_not_be_equal(first, second, msg, values)
 
-    def should_be_equal_as_strings(self, first, second, msg=None, values=True):
+    def should_be_equal_as_strings(self, first, second, msg=None, values=True, ignore_case=False):
         """Fails if objects are unequal after converting them to strings.
 
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``first`` and ``second`` should be compared case-insensitively,
+        provided that ``first`` and ``second`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         If both arguments are multiline strings, the comparison is done using
         `multiline string comparisons`.
         """
         self._log_types_at_info_if_different(first, second)
         first, second = [self._convert_to_string(i) for i in (first, second)]
+        if is_truthy(ignore_case) and is_string(first) and is_string(second):
+            first = first.lower()
+            second = second.lower()
         self._should_be_equal(first, second, msg, values)
 
-    def should_not_start_with(self, str1, str2, msg=None, values=True):
+    def should_not_start_with(self, str1, str2, msg=None, values=True, ignore_case=False):
         """Fails if the string ``str1`` starts with the string ``str2``.
 
+        ``ignore_case`` is False by default.  If True, it indicates that
+        ``str1`` and ``str2`` should be compared case-insensitively,
+        provided that ``str1`` and ``str2`` are string types.  See `Boolean
+        arguments` section for more details.  (New in RF 3.0.1)
+
         See `Should Be Equal` for an explanation on how to override the default
         error message with ``msg`` and ``values``.
         """
-        if str1.startswith(str2):
+        prv_str1 = str1
+        prv_str2 = str2
+        if is_truthy(ignore_case) and is_string(str1) and is_string(str2):
+            prv_str1 = str1.lower()
+            prv_str2 = str2.lower()
+        if prv_str1.startswith(prv_str2):
```

Comment:
I understand the point of avoiding exceptions, but there's anyway going to be an exception when `.startswith()` is run unless strings are used. The extra protection for `.lower()` thus doesn't help at all but adds a little complexity. If we wanted to add a guard to avoid invalid usage, we should check argument types in the beginning of the keyword and raise an exception if types are invalid. I don't think that's worth the effort with a keyword clearly expecting strings.

I also see the point to use the original strings in the error message, but that should then be done consistently with all keywords. Now, for example, `should_be_equal` just does `first = first.lower()`. This is something that we need to talk separately rather than in code comments.